waku/v2/protocol/waku_message.nim

```diff
@@ -41,9 +37,9 @@ type WakuMessage* = object
     # this field will be used in the rln-relay protocol
     # XXX Experimental, this is part of https://rfc.vac.dev/spec/17/ spec and not yet part of WakuMessage spec
     when defined(rln):
-      proof*: RateLimitProof
+      proof*: seq[byte]
```

Comment:
There was an initial intention to define `RateLimitProof` and not directly `seq[bytes]`, it is an alias and will make it swappable with other future types and easier and clearer for interfacing with other procs/modules. Also, it is inefficient to have recurring decoding/encoding when we can do the decoding once and keep using the resultant `RateLimitProof` object. 

An alternative approach to breaking the current circular dependency is: Given that the `WakuMessage` module only depends on the `RateLimitProof` data type, I suggest moving this type i.e., `RateLimitProof` out to a separate module like `proof_types` and then changing `WakuMessage` dependent to that module.

Reply:
I understand this was intentional from the beginning. But I don't understand the following:

> There was an initial intention to define RateLimitProof and not directly seq[bytes], **it is an alias and will make it swappable with other future types** and easier and clearer for interfacing with other procs/modules.

What do you mean by "it is an alias"? How do you expect to swap this?

----

> Also, it is inefficient to have recurring decoding/encoding when we can do the decoding once and keep using the resultant RateLimitProof object.

Regarding the state of the RLN code, performance in decoding is one of the less important things to worry about at this moment. Decoding a complex field, like `proof: RateLimitProof`, that is not used in other protocols adds overhead to the rest of the protocols.
 
Different strategies could be used to optimize the implementation (e.g., duplicate the WakuMessage type within the RLN module and decode in the Waku relay handler with the new type). But in any case, the optimization should be done once the code is in better shape.

Additionally, using a simple buffer (a.k.a. `seq[byte]`) opens the possibility to change the field format if the `RateLimitProof` format changes without changing it in a thousand places in the code. We will be able to modify the RLN code freely without having to touch common parts of the codebase. 

In other words, this approach is less risky and future-proof.

> An alternative approach to breaking the current circular dependency is: Given that the WakuMessage module only depends on the `RateLimitProof` data type, I suggest moving this type, i.e., `RateLimitProof` out to a separate module like `proof_types` and then changing WakuMessage dependent to that module.

Sorry, but this is not a good option. 

We are using the `waku_node` as a leave of the imports graph. All modules import the `waku_message`. So this module should not import any other module.

With that approach, we are adding unnecessary complexity (e.g., unnecessary dependencies between modules, scattering code that is not collocated around instead of keeping it within the module that uses it). 

This is one of my software principles for maintainable code:

> **All the code that changes together should be together.**